prusti-specs/src/extern_spec_rewriter.rs

```diff
@@ -169,20 +29,95 @@ fn rewrite_self(tokens: proc_macro2::TokenStream) -> proc_macro2::TokenStream {
     new_tokens
 }
 
-fn rewrite_method_inputs(item_ty: &syn::Type, method: &mut ImplItemMethod) ->
-    syn::punctuated::Punctuated<syn::Expr, syn::token::Comma> {
+/// Add `PhantomData` markers for each type parameter to silence errors
+/// about unused type parameters.
+///
+/// Given
+/// ```text
+/// struct Foo<A,B> {
+///    ...fields
+/// }
+/// ```
+/// Result
+/// ```text
+/// struct Foo<A,B> {
+///     ...fields
+///     ::core::marker::PhantomData<A>,
+///     ::core::marker::PhantomData<B>
+/// }
+/// ```
+fn add_phantom_data_for_generic_params(item_struct: &mut syn::ItemStruct) {
+    item_struct.generics = item_struct.generics.clone();
+    let generics = &item_struct.generics;
+
+    let mut fields: Vec<syn::Field> = Vec::new();
+    for param in generics.params.iter() {
+        let field: ParsableUnnamedField = match param {
+            syn::GenericParam::Type(tp) => {
+                let ident = tp.ident.clone();
+                syn::parse_quote! { ::core::marker::PhantomData<#ident> }
```

Comment:
Just wondering—do you really need to get a `syn` type here? Why not just `quote!` to get a `proc_macro2::TokenTree`, since in the end you are only using this to do `#(#fields),*`, i.e. tokenising. Then the `ParsableUnnamedField` type would not be necessary.